lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2232,20 +2233,30 @@ class SqlParser(private val ion: IonSystem) : Parser {
     /**
      * Parses a date string and validates that the date string is a string and of the format YYYY-MM-DD
      */
-    private fun List<Token>.parseDate(): ParseNode =
-        when (head?.value?.isText) {
-            true -> {
-                // validate that the date string follows the format YYYY-MM-DD
-                try {
-                    LocalDate.parse(head?.value?.stringValue())
-                } catch (e: DateTimeParseException) {
-                    err(e.localizedMessage, PARSE_INVALID_DATE_STRING)
-                }
-                ParseNode(DATE, head, listOf(), tail)
-            }
-            else -> err("Expected date string followed by the keyword DATE, found ${head?.value?.type}", PARSE_UNEXPECTED_TOKEN)
+    private fun List<Token>.parseDate(): ParseNode {
+        val dateStringToken = head
+        if (dateStringToken?.value == null || dateStringToken.type != LITERAL || !dateStringToken.value.isText) {
+            err("Expected date string followed by the keyword DATE, found ${head?.value?.type}", PARSE_UNEXPECTED_TOKEN)
         }
 
+        val dateString = dateStringToken.value.stringValue()
+
+        val datePatternRegex = Regex("\\d\\d\\d\\d-\\d\\d-\\d\\d")
```

Comment:
Another one, you can move this definition to a `private val` at the top of the file to avoid re-creating `Regex` instances you don't need.

Reply:
Done.